artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireConnection.java

```diff
@@ -768,7 +769,8 @@ public AMQConnectionContext initContext(ConnectionInfo info) throws Exception {
    }
 
    private void createInternalSession(ConnectionInfo info) throws Exception {
-      internalSession = server.createSession(UUIDGenerator.getInstance().generateStringUUID(), context.getUserName(), info.getPassword(), -1, this, true, false, false, false, null, null, true, operationContext, protocolManager.getPrefixes(), protocolManager.getSecurityDomain());
+      final String validatedUser = server.validateUser(context.getUserName(), info.getPassword(), this, protocolManager.getSecurityDomain());
```

Comment:
if OpenWireProtocolManager.validateUser could set a new field as OpenWireConnection.validatedUser the OpenWireConnection.createInternalSession shouldn't execute server.validateUser again

Reply:
agree